hkmc2/shared/src/main/scala/hkmc2/semantics/Elaborator.scala

```diff
@@ -469,6 +473,39 @@ extends Importer:
     // case _ =>
     //   ???
   
+  /** Module method applications that requrie further elaboration. */
+  def moduleMethod(t: Term): Ctxl[Term] = 
+  trace[Term](s"Elab (further) module method ${t.showDbg}", r => s"~> $r"):
+    def defn(t: Term, argLists: Ls[Term] = Nil): (Opt[Definition], Term, Ls[Term]) = t match
+      case Term.App(f, r) => defn(f, r :: argLists)
+      case Term.TyApp(f, _) => defn(f, argLists)
+      case _ => (t.symbol.flatMap(_.asBlkMember).flatMap(_.defn), t, argLists)
+    defn(t) match
+      case (S(defn: TermDefinition), inner, argLists) =>
+        log(s"Elab module method definition w/ type information ${defn}.")
+        val emptyTreeApp = new Tree.App(Tree.Empty(), Tree.Empty())
+        def unify(t: Term, paramLists: Ls[ParamList]): Term = 
```

Comment:
I'm not sure "unify" is the right term. In PL, that term already has a heavy proof-theoretic connotation. Why nit something like `zip`?

Reply:
Actually I chose *unify* because it is similar to what we did in the prolog terms unification, no? 🙈

Reply:
I'm not sure. Would be helpful if you could document, in the code, what this does. It's not obvious.

Reply:
I'll add some documentation. In brief, this piece of code takes a list of lists of parameters and a list of lists of arguments; comparing them one by one, and trying to insert some CtxArg if some is missing.

Reply:
Yeah that's zipping alright.